src/aura/CRLP_Rollup/CRLP_RollupHelper.js

```diff
@@ -59,282 +81,767 @@
             cmp.set("v.summaryFields", newFields);
-        } else if (context=='date') {
+        } else if (context === 'date') {
             newFields = this.filterFieldsByType(cmp, ["DATE"], newFields);
             cmp.set("v.dateFields", newFields);
-        } else if (context=='amount') {
+        } else if (context === 'amount') {
             newFields = this.filterFieldsByType(cmp, ["DOUBLE", "CURRENCY"], newFields);
             cmp.set("v.amountFields", newFields);
         }
     },
 
-    resetAllFields: function(cmp){
+    /* @description: resets all fields
+    */
+    resetAllFields: function (cmp) {
+        //todo: refactor to see if/when is even necessary
 
+        var summaryObject = cmp.get("v.activeRollup.Summary_Object__r.QualifiedApiName");
         var detailObject = cmp.get("v.activeRollup.Detail_Object__r.QualifiedApiName");
-        this.resetFields(cmp, detailObject, 'detail');
 
-        var summaryObject = cmp.get("v.activeRollup.Summary_Object__r.QualifiedApiName");
         this.resetFields(cmp, summaryObject, 'summary');
 
         //detail object is the provided object for amount fields because detail and amount object are ALWAYS the same
         this.resetFields(cmp, detailObject, 'amount');
+        this.resetFields(cmp, detailObject, 'detail');
 
         var dateObject = cmp.get("v.activeRollup.Date_Object__r.QualifiedApiName");
         this.resetFields(cmp, dateObject, 'date');
     },
-    filterFieldsByType: function(cmp, typeList, allFields) {
-        //if type is null, no detail field is selected
-        console.log("filter fields by type function");
-        var newFields = [];
 
-        typeList.forEach(function(type){
-            if (!(type == undefined || type == null)) {
-                allFields.forEach(function (field) {
-                    var datatype = field.type;
-                    if (datatype == type) {
-                        newFields.push(field);
-                    }
+    /* @description: resets the list of rollup types based on the selected summary object
+    */
+    resetRollupTypes: function (cmp) {
+        var summaryObject = cmp.get("v.activeRollup.Summary_Object__r.QualifiedApiName");
+        var labels = cmp.get("v.labels");
+        var templateList = [];
+        if (summaryObject === 'Account') {
+            templateList.push({
+                    label: labels.opportunityLabel + ' -> ' + labels.accountLabel + ' ' + labels.hardCredit,
+                    summaryObject: 'Account', name: 'Opportunity'
+                }
+                , {
+                    label: labels.paymentLabel + ' -> ' + labels.accountLabel + ' ' + labels.hardCredit
+                    , summaryObject: 'Account', name: 'npe01__OppPayment__c'
+                }
+                , {
+                    label: labels.opportunityLabel + ' -> ' + labels.accountLabel + ' ' + labels.softCredit
+                    , summaryObject: 'Account', name: 'Partial_Soft_Credit__c'
```

Comment:
Can you remind me of any Apex methods we have built out to ensure the namespace is correct for a given object? I can add a call to them into the UI_SVC class and add them to the labels map as the Name. There are probably a few other objects that will need this change as well.

Reply:
There are two approaches you can take:
1. `UTIL_Namespace.StrAllNSPrefix('Partial_Soft_Credit__c')`
2. `SObjectType.Partial_Soft_Credit__c.getName()`

Both will return the namespaced object when there is a package namespace in affect.

Reply:
Great, I think I'll use the latter since that's what's in use for the rest of the labels. Other than Partial_Soft_Credit__c, are there any other objects I should look out for? I don't think I'm naming rollups or filter groups directly in the UI, so those shouldn't be a problem.

Reply:
The two objects in the Cumulus package that will get the npsp namespace are Partial_Soft_Credit__c and General_Accounting_Unit__c.